_posts/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment.md

```diff
@@ -0,0 +1,396 @@
+- There are also manifests stored in ${Manifests} directory - [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test/files/manifests`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), for the cluster which can be used to deprovision the cluster at a later point in time.
+> **Note** : * the `manifests` directory is not present but created when you first use the scripts.*
+- Centos or Ubuntu images [are downloaded](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/tasks/download_image.yml) when provisioning a controlplane or a worker  
+
+<br/>
+<br/>
+
+
+
+
+
+!["An Overview of various resources generated while provisioning and their relationship amongst themselves"](/assets/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment/metal3-bmetal-arch-overview.png)  
+<br/>
+<br/>
+
+
+### Provision Cluster
+This script, located at the path - `${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh`, provisions the cluster by creating a `Metal3Cluster` resource. 
+
+<br/>
+To see if you have a successful Cluster resource creation( the cluster still doesn't have a Controlplane or Workers ), just do :
+
+```console
+kubectl get Metal3Cluster ${CLUSTER_NAME} -n metal3
+```
+> This will return the cluster deployed, and you can check the cluster details by describing the returned resource.
+
+<br/>
+Here is what a `Cluster` resource looks like :
+
+```console
+kubectl describe Cluster ${CLUSTER_NAME} -n metal3
+```
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+metadata:
+  [......]
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/18
+    services:
+      cidrBlocks:
+      - 10.96.0.0/12
+  controlPlaneEndpoint:
+    host: 192.168.111.249
+    port: 6443
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    kind: KubeadmControlPlane
+    name: bmetalcluster
+    namespace: metal3
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: Metal3Cluster
+    name: bmetalcluster
+    namespace: metal3
+status:
+  infrastructureReady: true
+  phase: Provisioned
+```
+  
+<br/>
+<br/>
+
+### Provision Controlplane
+
+This script, located at the path - `${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh`, provisions the controlplane or the master of the cluster. As part of the controlplane provisioning, the generated manifests are for cluster, controlplane and machinedeployment. The `MachineDeployment` creates a `Machine` which picks up a BareMetalHost satisfying its requirements as the controlplane node, and it is then provisioned by the BareMetalOperator. there are other resources created in the process of provisioning, like `KubeadmControlPlane` and `Metal3MachineTemplate` resource which are involved in the provisioning process.
```

Comment:
We should be consistent about capitalization of resource names. For example, "controlplane" should be either "control plane" when talking generally or ControlPlane when talking about the API resource. The same applies for Cluster and MachineDeployment.

"BareMetalOperator" should probably be "baremetal-operator", since that is the name of the repo where that component lives.

Reply:
Yes, I am working on fixing this